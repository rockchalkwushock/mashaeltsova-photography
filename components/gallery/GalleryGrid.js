const GalleryGrid = ({ children }) => (
  <div>
    {children}
    <style jsx>{`
      div {
        border: 2px solid lime;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(320px, auto));
        grid-gap: 0.5em;
        justify-items: center;
        padding-top: 0.5em;
      }
    `}</style>
  </div>
)

export default GalleryGrid
