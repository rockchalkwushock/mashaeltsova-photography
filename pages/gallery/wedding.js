import { PageWithoutData } from '../../lib'

export default PageWithoutData(() => <h1>Wedding</h1>)
