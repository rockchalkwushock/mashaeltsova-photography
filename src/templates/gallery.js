import React from 'react'
import Img from 'gatsby-image'

export const GalleryTemplate = ({ pathContext }) => (
  <div>
    <div>
      {pathContext.photos.map(({ node: photo }) => (
        <Img key={photo.id} resolutions={photo.resolutions} />
      ))}
    </div>
  </div>
)

export default GalleryTemplate
