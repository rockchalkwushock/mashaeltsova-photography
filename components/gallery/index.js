import GalleryGrid from './GalleryGrid'

export { GalleryGrid }
