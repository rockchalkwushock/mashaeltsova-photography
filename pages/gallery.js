import { Component } from 'react'
import Head from 'next/head'
import PropTypes from 'prop-types'
import NProgress from 'nprogress'
import { loadGetInitialProps } from 'next/dist/lib/utils'

import { A, Grid, Layout, Photo, SubNav, Thumb } from '../components'
import { fetchData, translate, withIntl } from '../lib'

export class Gallery extends Component {
  static propTypes = {
    photos: PropTypes.array.isRequired,
    intl: PropTypes.object.isRequired,
    locale: PropTypes.string.isRequired,
    messages: PropTypes.object.isRequired,
    url: PropTypes.shape({
      pathname: PropTypes.string.isRequired
    })
  }
  static async getInitialProps(ctx) {
    try {
      const props = await loadGetInitialProps(ctx)
      const { ids } = await fetchData('family')
      return { photos: ids, ...props }
    } catch (err) {
      throw err
    }
  }
  state = {
    currentView: 'family',
    error: false,
    image: null,
    images: null,
    message: '',
    modalPhoto: false
  }
  dismissModal = () => {
    this.setState(state => ({
      ...state,
      image: null,
      modalPhoto: false
    }))
  }
  handleOnClick = async e => {
    const target = translate(e.currentTarget.name)
    try {
      NProgress.start()
      const { error, ids } = await fetchData(target)
      if (error) {
        this.setState(state => ({
          ...state,
          currentView: target,
          error: true,
          image: null,
          images: ids,
          message: error,
          modalPhoto: false
        }))
        NProgress.done()
      }
      this.setState(state => ({
        ...state,
        currentView: target,
        image: null,
        images: ids,
        modalPhoto: false
      }))
      NProgress.done()
    } catch (err) {
      throw err
    }
  }
  showPhoto = id => {
    this.setState(state => ({
      ...state,
      image: id,
      modalPhoto: true
    }))
  }
  renderButtons = () =>
    this.props.messages.galleryNav
      .split(',')
      .map(t => (
        <A
          className="galleryButton"
          key={t}
          onClick={this.handleOnClick}
          text={t}
        />
      ))
  renderError = () => <h1>{this.state.message}</h1>
  renderView() {
    const imgs =
      this.state.currentView === 'family'
        ? this.props.photos
        : this.state.images
    if (this.state.modalPhoto) {
      return (
        <Photo
          onDismiss={() => this.dismissModal()}
          publicId={this.state.image}
        />
      )
    }
    return imgs.map(id => (
      <Thumb
        key={id}
        onClick={() => this.showPhoto(id)}
        publicId={id}
        view={this.state.currentView}
      />
    ))
  }
  render() {
    return (
      <Layout messages={this.props.messages} url={this.props.url}>
        <Head>
          <title>Masha Eltsova Photography | Gallery</title>
          {/* SEO */}
          <meta
            name="description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          {/* SCHEMA.ORG FOR GOOGLE */}
          <meta itemProp="name" content="Masha Eltsova Photography | Gallery" />
          <meta
            itemProp="description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          {/* TWITTER */}
          <meta
            name="twitter:title"
            content="Masha Eltsova Photography | Gallery"
          />
          <meta
            name="twitter:description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          <meta
            name="twitter:url"
            content="https://mashaeltsovaphotography.com/gallery"
          />
          {/* FACEBOOK, PINTREST, GOOGLE+ */}
          <meta
            property="og:title"
            content="Masha Eltsova Photography | Gallery"
          />
          <meta
            property="og:description"
            content="Masha Eltsova Photography professional gallery and portfolio."
          />
          <meta
            property="og:site_name"
            content="Masha Eltsova Photography | Gallery"
          />
          <meta
            property="og:url"
            content="https://mashaeltsovaphotography.com/gallery"
          />
        </Head>
        <SubNav>{this.renderButtons()}</SubNav>
        <Grid>{this.state.error ? this.renderError() : this.renderView()}</Grid>
      </Layout>
    )
  }
}

export default withIntl(Gallery)
