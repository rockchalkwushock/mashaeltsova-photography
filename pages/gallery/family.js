import Head from 'next/head'
import PropTypes from 'prop-types'

import { messages, PageWithoutData } from '../../lib'
import { Footer, Gallery, Layout } from '../../components'

/**
 * FIXME
 *
 * Create a messages[xyz] for the correct title/content/description.
 *
 * REVIEW
 *
 * this.props.messages will hold the messages; however,
 * this is in production ONLY. In development the object
 * is empty.
 *
 * QUESTION:
 *
 * 1) Can I re-write some code to supply the messages in
 * development as well? This would make importing the messages
 * object from '/lib' unnecessary.
 *
 * 2) Can I write some kind of 'if' block to use this.props.messages
 * if the above is not possible?
 */
const FamilyGallery = ({ ids, intl, url }) => (
  <Layout pathname={url.pathname}>
    <Head key={0}>
      <title>{intl.formatMessage(messages.title)}</title>
      <meta description={intl.formatMessage(messages.description)} />
    </Head>
    <Gallery key={1} ids={ids} pathname={url.pathname} />
    <Footer key={2} intl={intl} />
  </Layout>
)

FamilyGallery.propTypes = {
  ids: PropTypes.array.isRequired,
  intl: PropTypes.object.isRequired,
  messages: PropTypes.object.isRequired,
  url: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  })
}

export default PageWithoutData(FamilyGallery)
