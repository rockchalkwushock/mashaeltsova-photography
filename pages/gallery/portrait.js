import Head from 'next/head'
import PropTypes from 'prop-types'

import { messages, PageWithoutData } from '../../lib'
import { Footer, Gallery } from '../../components'

const PortraitGallery = ({ ids, intl, url }) => [
  <Head key={0}>
    <title>{intl.formatMessage(messages.title)}</title>
    <meta description={intl.formatMessage(messages.description)} />
  </Head>,
  <Gallery key={1} ids={ids} pathname={url.pathname} />,
  <Footer key={2} intl={intl} />
]

PortraitGallery.propTypes = {
  ids: PropTypes.array.isRequired,
  intl: PropTypes.object.isRequired,
  messages: PropTypes.object.isRequired,
  url: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  })
}

export default PageWithoutData(PortraitGallery)
