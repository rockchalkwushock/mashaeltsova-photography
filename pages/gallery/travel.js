import Head from 'next/head'
import PropTypes from 'prop-types'

import { messages, PageWithoutData } from '../../lib'
import { Footer, Gallery } from '../../components'

const TravelGallery = ({ ids, intl, url }) => [
  <Head key={0}>
    <title>{intl.formatMessage(messages.title)}</title>
    <meta description={intl.formatMessage(messages.description)} />
  </Head>,
  <Gallery key={1} ids={ids} pathname={url.pathname} />,
  <Footer key={2} intl={intl} />
]

TravelGallery.propTypes = {
  ids: PropTypes.array.isRequired,
  intl: PropTypes.object.isRequired,
  messages: PropTypes.object.isRequired,
  url: PropTypes.shape({
    pathname: PropTypes.string.isRequired
  })
}

export default PageWithoutData(TravelGallery)
