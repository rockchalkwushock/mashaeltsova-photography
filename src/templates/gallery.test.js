import React from 'react'
import { mount } from 'enzyme'

import { GalleryTemplate } from './gallery'

const pathContext = {
  photos: [
    {
      node: {
        id: 1,
        resolutions: {
          base64: '/static/pic.jpg',
          height: 200,
          src: '/static/pic.jpg',
          srcSet: '/static/pic.jpg',
          srcWebp: '/static/pic.jpg',
          srcSetWebp: '/static/pic.jpg',
          width: 200
        }
      }
    }
  ]
}

describe('Template: <Gallery />', () => {
  test('should render markup', () => {
    const wrapper = mount(<GalleryTemplate pathContext={pathContext} />)
    expect(wrapper).toMatchSnapshot()
  })
})
