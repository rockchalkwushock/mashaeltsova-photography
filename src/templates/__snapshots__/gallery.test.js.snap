// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template: <Gallery /> should render markup 1`] = `
<GalleryTemplate
  pathContext={
    Object {
      "photos": Array [
        Object {
          "node": Object {
            "id": 1,
            "resolutions": Object {
              "base64": "/static/pic.jpg",
              "height": 200,
              "src": "/static/pic.jpg",
              "srcSet": "/static/pic.jpg",
              "srcSetWebp": "/static/pic.jpg",
              "srcWebp": "/static/pic.jpg",
              "width": 200,
            },
          },
        },
      ],
    }
  }
>
  <div>
    <div>
      <Image
        Tag="div"
        alt=""
        fadeIn={true}
        key="1"
        resolutions={
          Object {
            "base64": "/static/pic.jpg",
            "height": 200,
            "src": "/static/pic.jpg",
            "srcSet": "/static/pic.jpg",
            "srcSetWebp": "/static/pic.jpg",
            "srcWebp": "/static/pic.jpg",
            "width": 200,
          }
        }
      >
        <div
          className=" gatsby-image-outer-wrapper"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className=" gatsby-image-wrapper"
            style={
              Object {
                "display": "inline-block",
                "height": 200,
                "overflow": "hidden",
                "position": "relative",
                "width": 200,
              }
            }
          >
            <Img
              alt=""
              src="/static/pic.jpg"
              style={
                Object {
                  "opacity": 0,
                  "transitionDelay": "0.25s",
                }
              }
            >
              <img
                alt=""
                src="/static/pic.jpg"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 0.5s",
                    "transitionDelay": "0.25s",
                    "width": "100%",
                  }
                }
              />
            </Img>
            <Img
              alt=""
              height={200}
              onLoad={[Function]}
              src="/static/pic.jpg"
              srcSet="/static/pic.jpg"
              style={
                Object {
                  "opacity": 1,
                }
              }
              width={200}
            >
              <img
                alt=""
                height={200}
                onLoad={[Function]}
                src="/static/pic.jpg"
                srcSet="/static/pic.jpg"
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 0.5s",
                    "width": "100%",
                  }
                }
                width={200}
              />
            </Img>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<img width=\\"200\\" height=\\"200\\" src=\\"/static/pic.jpg\\" srcset=\\"/static/pic.jpg\\" alt=\\"\\" title=\\"\\" sizes=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/>",
                }
              }
            />
          </div>
        </div>
      </Image>
    </div>
  </div>
</GalleryTemplate>
`;
